lib/jwt_signed_request.rb

```diff
@@ -77,7 +77,11 @@ def self.request_body(request:)
   private_class_method :request_body
 
   def self.verified_headers?(request:, claims:)
-    parsed_headers = JSON.parse(claims['headers'])
+    parsed_headers = begin
+      JSON.parse(claims['headers'])
+    rescue JSON::ParserError
+      {}
```

Comment:
Previously you could have passed an empty JSON hash and it would have passed the header verification. This simply allows you to pass nothing.

Taking another look I haven't properly handled the `nil` case, I'll be pushing another commit to handle that properly.

We could raise more specific errors if certain claims are missing but I would guess that could be giving more information to attackers.